applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_scan_for_conn.c

```diff
@@ -0,0 +1,170 @@
+		LOG_DBG("Found bonded device");
+
+		bt_le_scan_cb_unregister(&scan_callback);
+		cb_registered = false;
+
+		ret = bt_le_scan_stop();
+		if (ret) {
+			LOG_WRN("Stop scan failed: %d", ret);
+		}
+
+		ret = bt_conn_le_create(adv_addr, BT_CONN_LE_CREATE_CONN, CONNECTION_PARAMETERS,
+					&conn);
+		if (ret) {
+			LOG_WRN("Create ACL connection failed: %d", ret);
+
+			ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_CONN, NULL);
+			if (ret) {
+				LOG_WRN("Failed to restart scanning: %d", ret);
+			}
+		}
+	}
+}
+
+static bool device_found(struct bt_data *data, void *user_data)
+{
+	int ret;
+	bt_addr_le_t *addr = user_data;
+	struct bt_conn *conn;
+
+	/* Check type == COMPLETE_NAME */
+	if (data->type != BT_DATA_NAME_COMPLETE) {
+		return true;
+	}
+
+	if ((data->data_len == strlen(srch_name)) &&
+	    (strncmp(srch_name, data->data, strlen(srch_name)) == 0)) {
+		LOG_DBG("Device found");
+
+		bt_le_scan_cb_unregister(&scan_callback);
+		cb_registered = false;
+
+		ret = bt_le_scan_stop();
+		if (ret) {
+			LOG_WRN("Stop scan failed: %d", ret);
+		}
+
+		ret = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, CONNECTION_PARAMETERS, &conn);
+		if (ret) {
+			LOG_ERR("Could not init connection");
+
+			ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_CONN, NULL);
+			if (ret) {
+				LOG_WRN("Failed to restart scanning: %d", ret);
+			}
+		}
+
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * @brief      Scan receive callback when scanning for connections
+ *
+ * @param[in]  info  LE advertisement and scan response packet information
+ * @param      ad  Received advertising data
+ */
+static void scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad)
+{
+
+	/* We only care about connectable advertisers */
+	if (!(info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE)) {
+		return;
+	}
+
+	switch (info->adv_type) {
+	case BT_GAP_ADV_TYPE_ADV_DIRECT_IND:
+		/* Direct advertising has no payload, so no need to parse */
+		if (bonded_num) {
+			bt_foreach_bond(BT_ID_DEFAULT, bond_connect, (void *)info->addr);
+		}
+		break;
+	case BT_GAP_ADV_TYPE_ADV_IND:
+		/* Fall through */
+	case BT_GAP_ADV_TYPE_EXT_ADV:
+		/* Fall through */
+	case BT_GAP_ADV_TYPE_SCAN_RSP:
+		/* Note: May lead to connection creation */
+		if (bonded_num < CONFIG_BT_MAX_PAIRED) {
+
+			bt_data_parse(ad, device_found, (void *)info->addr);
+		}
+		break;
+	default:
+		break;
+	}
+}
+
+int bt_mgmt_scan_for_conn_start(struct bt_le_scan_param *scan_param, char const *const name)
+{
+	int ret;
+
+	srch_name = name;
+
+	if (!cb_registered) {
+		scan_callback.recv = scan_recv;
+		bt_le_scan_cb_register(&scan_callback);
+		cb_registered = true;
+	} else {
+		LOG_WRN("Already scanning");
+		return -EAGAIN;
+	}
+
+	/* Reset number of bonds found */
+	bonded_num = 0;
+
+	bt_foreach_bond(BT_ID_DEFAULT, bond_check, NULL);
+
+	if (bonded_num >= CONFIG_BT_MAX_PAIRED) {
+		LOG_INF("All bonded slots filled, will not accept new devices");
```

Comment:
Vote that this is changed to a WRN. It may be confusing when this happens.

Reply:
Disagree, since this is printed everytime a bonded device disconnects (after the bonded slots are filled)